src/compiler/interface.jl

```diff
@@ -40,9 +40,13 @@ _pullback(f, args...) = _pullback(Context(), f, args...)
 tailmemaybe(::Nothing) = nothing
 tailmemaybe(x::Tuple) = Base.tail(x)
 
+replacezero(t) = ntuple(i -> t[i] isa AbstractZero ? nothing : t[i], length(t))
+replacezero(::Nothing) = nothing
+replacezero(::AbstractZero) = nothing
+
 function pullback(f, args...)
   y, back = _pullback(f, args...)
-  y, Δ -> tailmemaybe(back(Δ))
+  y, Δ -> replacezero(tailmemaybe(back(Δ)))
```

Comment:
Maybe rather than `replacezero` we could just call the existing  `wrap_chainrules_output`
https://github.com/FluxML/Zygote.jl/blob/master/src/compiler/chainrules.jl#L42

Reply:
I tried it but it compresses a tuple of `Zero()` into a single `nothing` which is not desired here. Also it does not solve the nested AD issue

Reply:
A tuple of `Zero`s should be nothing by Zygote's conventions I think.

Reply:
Revisited this and made it work now, some minor tweaks were needed.